quadlets/adguard/adguard.container

```diff
@@ -0,0 +1,28 @@
+[Unit]
+Description=Adguard Quadlet
+
+[Service]
+Restart=always
+TimeoutStartSec=900
+
+[Install]
+WantedBy=default.target
+
+[Container]
+Image=docker.io/adguard/adguardhome:latest
+ContainerName=adguard
+HostName=adguard
+
+# Optionally run this on network mode host, if you want real client ips to show in the logs
```

Comment:
Is this problem solved after `passt` or `pasta` becomes the default? Right now I'm running with `slirp4netns` because on AlmaLinux my podman is still on 4.x. I actually have compiled 5.x from source to test socket activation, so I tried `pasta` a bit. I know it works, but I have to finesse it on boot to start my containers because something comes up in the wrong order and there's no network when the containers try to start. I'm really not sure what all was going wrong yet. I think on a distro like fedora instead it might be working normally.

Anyway, it's supposed to support source ip mapping. Are you running a version of podman/adguard/pasta where we could try that out? Maybe it can get rid of this note.

Reply:
I will test this today I have a load balanced setup so I can test on my failover node.

Reply:
```
[
     {
          "Id": "d5e3bd1a56ed9e03aa28a686d9fd68cabd4a54863fa5e3448f3fa3436d19fe01",
          "Created": "2024-12-05T09:00:13.467515789Z",
          "Path": "/opt/adguardhome/AdGuardHome",
          "Args": [
               "--no-check-update",
               "-c",
               "/opt/adguardhome/conf/AdGuardHome.yaml",
               "-w",
               "/opt/adguardhome/work"
          ],
          "State": {
               "OciVersion": "1.2.0",
               "Status": "running",
               "Running": true,
               "Paused": false,
               "Restarting": false,
               "OOMKilled": false,
               "Dead": false,
               "Pid": 46948,
               "ConmonPid": 46946,
               "ExitCode": 0,
               "Error": "",
               "StartedAt": "2024-12-05T09:00:13.784008714Z",
               "FinishedAt": "0001-01-01T00:00:00Z",
               "CgroupPath": "/user.slice/user-1000.slice/user@1000.service/app.slice/adguard.service/libpod-payload-d5e3bd1a56ed9e03aa28a686d9fd68cabd4a54863fa5e3448f3fa3436d19fe01",
               "CheckpointedAt": "0001-01-01T00:00:00Z",
               "RestoredAt": "0001-01-01T00:00:00Z"
          },
          "Image": "8514b103589293aa5a5dc3e923e12709e65eadfb735ee6a69fc467693c6ac974",
          "ImageDigest": "sha256:ec59d9d8d083b74620f827879bee7ad88621f96bc0c4347c84c176ffa349484c",
          "ImageName": "docker.io/adguard/adguardhome:latest",
          "Rootfs": "",
          "Pod": "",
          "ResolvConfPath": "/run/user/1000/containers/overlay-containers/d5e3bd1a56ed9e03aa28a686d9fd68cabd4a54863fa5e3448f3fa3436d19fe01/userdata/resolv.conf",
          "HostnamePath": "/run/user/1000/containers/overlay-containers/d5e3bd1a56ed9e03aa28a686d9fd68cabd4a54863fa5e3448f3fa3436d19fe01/userdata/hostname",
          "HostsPath": "/run/user/1000/containers/overlay-containers/d5e3bd1a56ed9e03aa28a686d9fd68cabd4a54863fa5e3448f3fa3436d19fe01/userdata/hosts",
          "StaticDir": "/home/jphadmin/.local/share/containers/storage/overlay-containers/d5e3bd1a56ed9e03aa28a686d9fd68cabd4a54863fa5e3448f3fa3436d19fe01/userdata",
          "OCIConfigPath": "/home/jphadmin/.local/share/containers/storage/overlay-containers/d5e3bd1a56ed9e03aa28a686d9fd68cabd4a54863fa5e3448f3fa3436d19fe01/userdata/config.json",
          "OCIRuntime": "crun",
          "ConmonPidFile": "/run/user/1000/containers/overlay-containers/d5e3bd1a56ed9e03aa28a686d9fd68cabd4a54863fa5e3448f3fa3436d19fe01/userdata/conmon.pid",
          "PidFile": "/run/user/1000/containers/overlay-containers/d5e3bd1a56ed9e03aa28a686d9fd68cabd4a54863fa5e3448f3fa3436d19fe01/userdata/pidfile",
          "Name": "adguard",
          "RestartCount": 0,
          "Driver": "overlay",
          "MountLabel": "system_u:object_r:container_file_t:s0:c263,c384",
          "ProcessLabel": "system_u:system_r:container_t:s0:c263,c384",
          "AppArmorProfile": "",
          "EffectiveCaps": [
               "CAP_CHOWN",
               "CAP_DAC_OVERRIDE",
               "CAP_FOWNER",
               "CAP_FSETID",
               "CAP_KILL",
               "CAP_NET_BIND_SERVICE",
               "CAP_SETFCAP",
               "CAP_SETGID",
               "CAP_SETPCAP",
               "CAP_SETUID",
               "CAP_SYS_CHROOT"
          ],
          "BoundingCaps": [
               "CAP_CHOWN",
               "CAP_DAC_OVERRIDE",
               "CAP_FOWNER",
               "CAP_FSETID",
               "CAP_KILL",
               "CAP_NET_BIND_SERVICE",
               "CAP_SETFCAP",
               "CAP_SETGID",
               "CAP_SETPCAP",
               "CAP_SETUID",
               "CAP_SYS_CHROOT"
          ],
          "ExecIDs": [],
          "GraphDriver": {
               "Name": "overlay",
               "Data": {
                    "LowerDir": "/home/jphadmin/.local/share/containers/storage/overlay/b03c2e14a6a7c82e46df24ace11fb4d6e1bc3351e65dc121232de11dbefdaf5e/diff:/home/jphadmin/.local/share/containers/storage/overlay/87bf2b96e6b648367936e4bc3326133b0123a55d9bf292f230053bbdd360bd19/diff:/home/jphadmin/.local/share/containers/storage/overlay/0b0c581ffd575308e0633508ba8ebb581ce24cc2fdfa9f86210fbc4750c80219/diff:/home/jphadmin/.local/share/containers/storage/overlay/0196c5c1b0792beac37d994da238c1d7f323cc52f405561f7c81498ca1e8656a/diff:/home/jphadmin/.local/share/containers/storage/overlay/4aeaa0e7d41fd419e3152bd151de17bcdd813a26287429b91503d1422a9470aa/diff",
                    "MergedDir": "/home/jphadmin/.local/share/containers/storage/overlay/e1c8a6f18853c8f1ed636b189c2dd30575bebf345e525939b0c75b0a3514a693/merged",
                    "UpperDir": "/home/jphadmin/.local/share/containers/storage/overlay/e1c8a6f18853c8f1ed636b189c2dd30575bebf345e525939b0c75b0a3514a693/diff",
                    "WorkDir": "/home/jphadmin/.local/share/containers/storage/overlay/e1c8a6f18853c8f1ed636b189c2dd30575bebf345e525939b0c75b0a3514a693/work"
               }
          },
          "Mounts": [
               {
                    "Type": "volume",
                    "Name": "adguard-work",
                    "Source": "/home/jphadmin/.local/share/containers/storage/volumes/adguard-work/_data",
                    "Destination": "/opt/adguardhome/work",
                    "Driver": "local",
                    "Mode": "z",
                    "Options": [
                         "nosuid",
                         "nodev",
                         "rbind"
                    ],
                    "RW": true,
                    "Propagation": "rprivate"
               },
               {
                    "Type": "volume",
                    "Name": "adguard-config",
                    "Source": "/home/jphadmin/.local/share/containers/storage/volumes/adguard-config/_data",
                    "Destination": "/opt/adguardhome/conf",
                    "Driver": "local",
                    "Mode": "z",
                    "Options": [
                         "nosuid",
                         "nodev",
                         "rbind"
                    ],
                    "RW": true,
                    "Propagation": "rprivate"
               },
               {
                    "Type": "bind",
                    "Source": "/var/log/AdGuardHome.log",
                    "Destination": "/var/log/AdGuardHome.log",
                    "Driver": "",
                    "Mode": "",
                    "Options": [
                         "rbind"
                    ],
                    "RW": true,
                    "Propagation": "rprivate"
               }
          ],
          "Dependencies": [],
          "NetworkSettings": {
               "EndpointID": "",
               "Gateway": "",
               "IPAddress": "",
               "IPPrefixLen": 0,
               "IPv6Gateway": "",
               "GlobalIPv6Address": "",
               "GlobalIPv6PrefixLen": 0,
               "MacAddress": "",
               "Bridge": "",
               "SandboxID": "",
               "HairpinMode": false,
               "LinkLocalIPv6Address": "",
               "LinkLocalIPv6PrefixLen": 0,
               "Ports": {
                    "3000/tcp": [
                         {
                              "HostIp": "0.0.0.0",
                              "HostPort": "3000"
                         }
                    ],
                    "3000/udp": null,
                    "443/tcp": [
                         {
                              "HostIp": "0.0.0.0",
                              "HostPort": "8443"
                         }
                    ],
                    "443/udp": null,
                    "53/tcp": [
                         {
                              "HostIp": "0.0.0.0",
                              "HostPort": "53"
                         }
                    ],
                    "53/udp": [
                         {
                              "HostIp": "0.0.0.0",
                              "HostPort": "53"
                         }
                    ],
                    "5443/tcp": null,
                    "5443/udp": null,
                    "6060/tcp": null,
                    "67/udp": null,
                    "68/udp": null,
                    "784/udp": [
                         {
                              "HostIp": "0.0.0.0",
                              "HostPort": "784"
                         }
                    ],
                    "80/tcp": [
                         {
                              "HostIp": "0.0.0.0",
                              "HostPort": "8844"
                         }
                    ],
                    "853/tcp": [
                         {
                              "HostIp": "0.0.0.0",
                              "HostPort": "853"
                         }
                    ],
                    "853/udp": null
               },
               "SandboxKey": "/run/user/1000/netns/netns-16cb5875-7dae-a547-4eca-5e9390b22876"
          },
          "Namespace": "",
          "IsInfra": false,
          "IsService": false,
          "KubeExitCodePropagation": "invalid",
          "lockNumber": 0,
          "Config": {
               "Hostname": "adguard",
               "Domainname": "",
               "User": "",
               "AttachStdin": false,
               "AttachStdout": false,
               "AttachStderr": false,
               "Tty": false,
               "OpenStdin": false,
               "StdinOnce": false,
               "Env": [
                    "container=podman",
                    "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                    "HOSTNAME=adguard",
                    "HOME=/root"
               ],
               "Cmd": [
                    "--no-check-update",
                    "-c",
                    "/opt/adguardhome/conf/AdGuardHome.yaml",
                    "-w",
                    "/opt/adguardhome/work"
               ],
               "Image": "docker.io/adguard/adguardhome:latest",
               "Volumes": null,
               "WorkingDir": "/opt/adguardhome/work",
               "Entrypoint": [
                    "/opt/adguardhome/AdGuardHome"
               ],
               "OnBuild": null,
               "Labels": {
                    "PODMAN_SYSTEMD_UNIT": "adguard.service",
                    "maintainer": "AdGuard Team \u003cdevteam@adguard.com\u003e",
                    "org.opencontainers.image.authors": "AdGuard Team \u003cdevteam@adguard.com\u003e",
                    "org.opencontainers.image.created": "2024-11-06T13:21:16Z",
                    "org.opencontainers.image.description": "Network-wide ads \u0026 trackers blocking DNS server",
                    "org.opencontainers.image.documentation": "https://github.com/AdguardTeam/AdGuardHome/wiki/",
                    "org.opencontainers.image.licenses": "GPL-3.0",
                    "org.opencontainers.image.revision": "54f3a5f990d6415ff7fafddca2f4848383cd3867",
                    "org.opencontainers.image.source": "https://github.com/AdguardTeam/AdGuardHome",
                    "org.opencontainers.image.title": "AdGuard Home",
                    "org.opencontainers.image.url": "https://adguard.com/en/adguard-home/overview.html",
                    "org.opencontainers.image.vendor": "AdGuard",
                    "org.opencontainers.image.version": "v0.107.54"
               },
               "Annotations": {
                    "io.container.manager": "libpod",
                    "io.podman.annotations.autoremove": "TRUE",
                    "io.podman.annotations.cid-file": "/run/user/1000/adguard.cid",
                    "org.opencontainers.image.stopSignal": "15",
                    "org.systemd.property.KillSignal": "15",
                    "org.systemd.property.TimeoutStopUSec": "uint64 10000000"
               },
               "StopSignal": "SIGTERM",
               "HealthcheckOnFailureAction": "none",
               "CreateCommand": [
                    "/usr/bin/podman",
                    "run",
                    "--name=adguard",
                    "--cidfile=/run/user/1000/adguard.cid",
                    "--replace",
                    "--rm",
                    "--cgroups=split",
                    "--sdnotify=conmon",
                    "-d",
                    "-v",
                    "/var/log/AdGuardHome.log:/var/log/AdGuardHome.log:z",
                    "-v",
                    "adguard-work:/opt/adguardhome/work:z",
                    "-v",
                    "adguard-config:/opt/adguardhome/conf:z",
                    "--publish",
                    "53:53/tcp",
                    "--publish",
                    "53:53/udp",
                    "--publish",
                    "784:784/udp",
                    "--publish",
                    "853:853/tcp",
                    "--publish",
                    "3000:3000/tcp",
                    "--publish",
                    "8844:80/tcp",
                    "--publish",
                    "8443:443/tcp",
                    "--hostname",
                    "adguard",
                    "docker.io/adguard/adguardhome:latest"
               ],
               "Umask": "0022",
               "Timeout": 0,
               "StopTimeout": 10,
               "Passwd": true,
               "sdNotifyMode": "conmon",
               "sdNotifySocket": "/run/user/1000/systemd/notify"
          },
          "HostConfig": {
               "Binds": [
                    "adguard-work:/opt/adguardhome/work:z,rw,rprivate,nosuid,nodev,rbind",
                    "adguard-config:/opt/adguardhome/conf:z,rw,rprivate,nosuid,nodev,rbind",
                    "/var/log/AdGuardHome.log:/var/log/AdGuardHome.log:rw,rprivate,rbind"
               ],
               "CgroupManager": "systemd",
               "CgroupMode": "private",
               "ContainerIDFile": "/run/user/1000/adguard.cid",
               "LogConfig": {
                    "Type": "journald",
                    "Config": null,
                    "Path": "",
                    "Tag": "",
                    "Size": "0B"
               },
               "NetworkMode": "pasta",
               "PortBindings": {
                    "3000/tcp": [
                         {
                              "HostIp": "0.0.0.0",
                              "HostPort": "3000"
                         }
                    ],
                    "443/tcp": [
                         {
                              "HostIp": "0.0.0.0",
                              "HostPort": "8443"
                         }
                    ],
                    "53/tcp": [
                         {
                              "HostIp": "0.0.0.0",
                              "HostPort": "53"
                         }
                    ],
                    "53/udp": [
                         {
                              "HostIp": "0.0.0.0",
                              "HostPort": "53"
                         }
                    ],
                    "784/udp": [
                         {
                              "HostIp": "0.0.0.0",
                              "HostPort": "784"
                         }
                    ],
                    "80/tcp": [
                         {
                              "HostIp": "0.0.0.0",
                              "HostPort": "8844"
                         }
                    ],
                    "853/tcp": [
                         {
                              "HostIp": "0.0.0.0",
                              "HostPort": "853"
                         }
                    ]
               },
               "RestartPolicy": {
                    "Name": "no",
                    "MaximumRetryCount": 0
               },
               "AutoRemove": true,
               "AutoRemoveImage": false,
               "Annotations": {
                    "io.container.manager": "libpod",
                    "io.podman.annotations.autoremove": "TRUE",
                    "io.podman.annotations.cid-file": "/run/user/1000/adguard.cid",
                    "org.opencontainers.image.stopSignal": "15",
                    "org.systemd.property.KillSignal": "15",
                    "org.systemd.property.TimeoutStopUSec": "uint64 10000000"
               },
               "VolumeDriver": "",
               "VolumesFrom": null,
               "CapAdd": [],
               "CapDrop": [],
               "Dns": [],
               "DnsOptions": [],
               "DnsSearch": [],
               "ExtraHosts": [],
               "GroupAdd": [],
               "IpcMode": "shareable",
               "Cgroup": "",
               "Cgroups": "default",
               "Links": null,
               "OomScoreAdj": 0,
               "PidMode": "private",
               "Privileged": false,
               "PublishAllPorts": false,
               "ReadonlyRootfs": false,
               "SecurityOpt": [],
               "Tmpfs": {},
               "UTSMode": "private",
               "UsernsMode": "",
               "ShmSize": 65536000,
               "Runtime": "oci",
               "ConsoleSize": [
                    0,
                    0
               ],
               "Isolation": "",
               "CpuShares": 0,
               "Memory": 0,
               "NanoCpus": 0,
               "CgroupParent": "",
               "BlkioWeight": 0,
               "BlkioWeightDevice": null,
               "BlkioDeviceReadBps": null,
               "BlkioDeviceWriteBps": null,
               "BlkioDeviceReadIOps": null,
               "BlkioDeviceWriteIOps": null,
               "CpuPeriod": 0,
               "CpuQuota": 0,
               "CpuRealtimePeriod": 0,
               "CpuRealtimeRuntime": 0,
               "CpusetCpus": "",
               "CpusetMems": "",
               "Devices": [],
               "DiskQuota": 0,
               "KernelMemory": 0,
               "MemoryReservation": 0,
               "MemorySwap": 0,
               "MemorySwappiness": 0,
               "OomKillDisable": false,
               "PidsLimit": 2048,
               "Ulimits": [
                    {
                         "Name": "RLIMIT_NOFILE",
                         "Soft": 524288,
                         "Hard": 524288
                    },
                    {
                         "Name": "RLIMIT_NPROC",
                         "Soft": 29652,
                         "Hard": 29652
                    }
               ],
               "CpuCount": 0,
               "CpuPercent": 0,
               "IOMaximumIOps": 0,
               "IOMaximumBandwidth": 0,
               "CgroupConf": null
          }
     }
]
```

`podman version 5.2.2`

If you omit the `Network` completely and expose the ports as normal (note I changed 80 to something else as its in use on the host) it works and shows up correctly as you can see it defaults to pasta networking and is fully operational.

Reply:
You already submitted the change for this, so I'll just take your updated version. I didn't notice you had put that together before I got up this morning.